msrest/pipeline/universal.py

```diff
@@ -129,10 +130,9 @@ def on_response(self, request, response, **kwargs):
 
 class RawDeserializer(SansIOHTTPPolicy):
 
-    JSON_MIMETYPES = [
-        'application/json',
-        'text/json' # Because we're open minded people...
-    ]
+    # Accept "text" because we're open minded people...
+    JSON_REGEXP = re.compile(r'(application|text)/([a-z+.]+\+)?json')
```

Comment:
This still won't match content types with parameters: 

Example: `Content-type: application/json; charset=utf-8`

Not blocking the PR for this. Just letting you know...

Reply:
Unless the parameters have already been stripped off (which it looks like they may have been at this point)

Reply:
Yes, they have at this point (see a little later in the file).